payment/src/main/java/org/killbill/billing/payment/api/DefaultApiBase.java

```diff
@@ -79,6 +80,13 @@ protected void logAPICall(final String transactionType, final Account account, f
                 logLine.append(", currency = ")
                        .append(currency);
             }
+
+            final String requestId = Request.getPerThreadRequestData() != null
```

Comment:
At the killbill-api level, the user token is created by the client and passed in the call context. At the JAX-RS level, it is automatically generated for the client.

#409 is about setting the user token to the request id, if specified.